### ISSUE TITLE
Intermittent_VS Code_Office invalid connection.

### ISSUE THREAD

**Issue by v-prakumar** [Azure/LogicAppsUX#2703]
### Describe the Bug with repro steps

1.Create a Stateful workflow in project.
2.Add when a email arrive as a trigger.
3.Save and Debug it.
4.After completion the run successfully go back to the Designer.

Expected: The connection should be valid.

Actual: The connection is invalid.
![image](https://github.com/Azure/LogicAppsUX/assets/92098490/d0334213-a204-4b70-aa81-29c6753dee9a)

### What type of Logic App Is this happening in?

Standard (Portal)

### Are you using Preview Designer or GA Designer

Preview

### Did you refer to the TSG before filing this issue? https://aka.ms/lauxtsg

No

### Workflow JSON

```json
{
    "definition": {
        "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
        "actions": {},
        "contentVersion": "1.0.0.0",
        "outputs": {},
        "triggers": {
            "When_a_new_email_arrives_(V3)": {
                "type": "ApiConnectionNotification",
                "inputs": {
                    "host": {
                        "connection": {
                            "referenceName": "office365"
                        }
                    },
                    "fetch": {
                        "pathTemplate": {
                            "template": "/v3/Mail/OnNewEmail"
                        },
                        "method": "get",
                        "queries": {
                            "importance": "Any",
                            "fetchOnlyWithAttachment": false,
                            "includeAttachments": false,
                            "folderPath": "Inbox"
                        }
                    },
                    "subscribe": {
                        "body": {
                            "NotificationUrl": "@{listCallbackUrl()}"
                        },
                        "pathTemplate": {
                            "template": "/GraphMailSubscriptionPoke/$subscriptions"
                        },
                        "method": "post",
                        "queries": {
                            "importance": "Any",
                            "fetchOnlyWithAttachment": false,
                            "folderPath": "Inbox"
                        }
                    }
                },
                "splitOn": "@triggerBody()?['value']"
            }
        }
    },
    "kind": "Stateful"
}
```

### Screenshots or Videos

****Local setting.json:** : Issue is repro while doing the Webhook testing as of now.**

{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "APP_KIND": "workflowapp",
    "ProjectDirectoryPath": "c:\\Users\\Administrator\\Desktop\\web",
    "FUNCTIONS_WORKER_RUNTIME": "node",
    "Workflows.WebhookRedirectHostUri": "https://490e-43-230-197-37.in.ngrok.io",
    "FUNCTIONS_EXTENSIONBUNDLE_SOURCE_URI": "https://cdnforlogicappsv2.blob.core.windows.net/logicapps-vendortesting",
    "WORKFLOWS_TENANT_ID": "72f988bf-86f1-41af-91ab-2d7cd011db47",
    "WORKFLOWS_SUBSCRIPTION_ID": "53ce23c6-ff79-490b-a7fb-9590fe7a414a",
    "WORKFLOWS_RESOURCE_GROUP_NAME": "First31",
    "WORKFLOWS_LOCATION_NAME": "centralus",
    "WORKFLOWS_MANAGEMENT_BASE_URI": "https://management.azure.com/",
    "office365-connectionKey": "eyJhbGciOiJSUzI1NiIsImtpZCI6IkFBREUxQTk4RUFFOTVGMzM3NzYwOEEyNzU2OUY5NUQ0RTVCOTk1RDIiLCJ4NXQiOiJxdDRhbU9ycFh6TjNZSW9uVnAtVjFPVzVsZEkiLCJ0eXAiOiJKV1QifQ.eyJ0cyI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsImNzIjoibG9naWMtYXBpcy1jZW50cmFsdXMvb2ZmaWNlMzY1L2E1MTBjNTQxY2ZhYjRiZjA4Yjc2ZjFiODBmNDk3ZDcxIiwidiI6Ii05MjIzMzcyMDM2ODU0Nzc1ODA4IiwiZW52aWQiOiI3ODFiOTc1M2FjMjVmODQxIiwiYXVkIjoiaHR0cHM6Ly83ODFiOTc1M2FjMjVmODQxLjAzLmNvbW1vbi5sb2dpYy1jZW50cmFsdXMuYXp1cmUtYXBpaHViLm5ldC9hcGltL29mZmljZTM2NS9hNTEwYzU0MWNmYWI0YmYwOGI3NmYxYjgwZjQ5N2Q3MSIsIm1hbmFnZW1lbnQiOiJodHRwczovL21hbmFnZW1lbnQubG9naWMtY2VudHJhbHVzLmF6dXJlLWFwaWh1Yi5uZXQvIiwibmJmIjoxNjg2MTU0NDQ5LCJleHAiOjE2ODY3NTkyNDksImlhdCI6MTY4NjE1NDQ0OSwiaXNzIjoiaHR0cHM6Ly9sb2dpYy1hcGlzLWNlbnRyYWx1cy5henVyZS1hcGltLm5ldC8ifQ.RzurpmIXCyWgdq4mIvBhFFkCT9czUAFdeqdhcbq7VDJXHxJ2tyYPo9Xs4abJlYRzm6niNmtU_XksVZ8Eox_rV23mIO6365fl4ATEozVfO8pghajre6Tv7aPqtjrPxaWwQKzhOfYFFyJvYrrIE4Zf3HEtIp8sbqIBNN8M8IuT6o7O4yDonrsM_5N911AS1biBnhvnXGxRCjbwEeDJf3sNkKod2beusgynAr0Ui-UmzY5nNHDCICGvOuaochAwt65Jc5s3eH9ctn_Gm7xpR3KXa3RHjRJG3DuJd5WTH7quFM9CVw-G20Y3HoWJoKq-zm-30v1PQlRC32oACvq8XaYskg",
    "Workflows.Sql.ConnectionString": "Server=tcp:portaluseserverpra.database.windows.net,1433;Initial Catalog=portalusedatabasepra;Persist Security Info=False;User ID=v-prakumar;Password=Shivani@9012700#;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;"
  }
}

**Connections.json:**

{
  "managedApiConnections": {
    "office365": {
      "api": {
        "id": "/subscriptions/53ce23c6-ff79-490b-a7fb-9590fe7a414a/providers/Microsoft.Web/locations/centralus/managedApis/office365"
      },
      "connection": {
        "id": "/subscriptions/53ce23c6-ff79-490b-a7fb-9590fe7a414a/resourceGroups/First31/providers/Microsoft.Web/connections/office365"
      },
      "connectionRuntimeUrl": "https://781b9753ac25f841.03.common.logic-centralus.azure-apihub.net/apim/office365/a510c541cfab4bf08b76f1b80f497d71/",
      "authentication": {
        "type": "Raw",
        "scheme": "Key",
        "parameter": "@appsetting('office365-connectionKey')"
      }
    }
  }
}

### Browser

VS CODE

### Additional context

1.VS Code extension: [vscode-azurelogicapps-2.24.3.vsix](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmicrosoft.sharepoint.com%2F%3Au%3A%2Ft%2FLogicApps%2FEUtS0SJECrFDgEfY_BSfFlsBSqFEmGM2jMuz1uN88uqHrQ%3Fe%3D62D7LG&data=05%7C01%7Cv-govubilish%40microsoft.com%7Ca7d1bc0a237d488d10e108db672b1080%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638217206935590829%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WlxG5bcq1ej761KNqQ%2BLtvAmRSTEgwvtejbZZlzSxZE%3D&reserved=0)
2.NuGet package: [https://artprodeussu2.artifacts.visualstudio.com/A41bf5486-7392-4b7a-a7e3-a735c767e3b3/b32aa71e-8ed2-41b2-9d77-5bc261222004/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL21zYXp1cmUvcHJvamVjdElkL2IzMmFhNzFlLThlZDItNDFiMi05ZDc3LTViYzI2MTIyMjAwNC9idWlsZElkLzc0Mzk0ODczL2FydGlmYWN0TmFtZS9kcm9wX2J1aWxkX0J1aWxkQXJ0aWZhY3RzX1dpbmRvd3M1/content?format=file&subPath=%2FMicrosoft.Azure.Workflows.WebJobs.Extension.1.25.4.nupkg](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fartprodeussu2.artifacts.visualstudio.com%2FA41bf5486-7392-4b7a-a7e3-a735c767e3b3%2Fb32aa71e-8ed2-41b2-9d77-5bc261222004%2F_apis%2Fartifact%2FcGlwZWxpbmVhcnRpZmFjdDovL21zYXp1cmUvcHJvamVjdElkL2IzMmFhNzFlLThlZDItNDFiMi05ZDc3LTViYzI2MTIyMjAwNC9idWlsZElkLzc0Mzk0ODczL2FydGlmYWN0TmFtZS9kcm9wX2J1aWxkX0J1aWxkQXJ0aWZhY3RzX1dpbmRvd3M1%2Fcontent%3Fformat%3Dfile%26subPath%3D%252FMicrosoft.Azure.Workflows.WebJobs.Extension.1.25.4.nupkg&data=05%7C01%7Cv-govubilish%40microsoft.com%7Ca7d1bc0a237d488d10e108db672b1080%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638217206935434622%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=itFALTueZQxguLRJvKKiWJXDyumAl4DtSnkg82YmPt8%3D&reserved=0)
3.Extension Bundle (CDN URL): [https://cdnforlogicappsv2.blob.core.windows.net/logicapps-vendortesting](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcdnforlogicappsv2.blob.core.windows.net%2Flogicapps-vendortesting&data=05%7C01%7Cv-govubilish%40microsoft.com%7Ca7d1bc0a237d488d10e108db672b1080%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638217206935590829%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=EuLLscGjv5qqENJ91h2yYlsC9J3u0tqkraH9d%2BPhXVg%3D&reserved=0)
4.Extension Bundle (Direct Download): [https://cdnforlogicappsv2.blob.core.windows.net/logicapps-vendortesting/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle.Workflows/1.25.4/Microsoft.Azure.Functions.ExtensionBundle.Workflows.1.25.4_any-any.zip](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcdnforlogicappsv2.blob.core.windows.net%2Flogicapps-vendortesting%2FExtensionBundles%2FMicrosoft.Azure.Functions.ExtensionBundle.Workflows%2F1.25.4%2FMicrosoft.Azure.Functions.ExtensionBundle.Workflows.1.25.4_any-any.zip&data=05%7C01%7Cv-govubilish%40microsoft.com%7Ca7d1bc0a237d488d10e108db672b1080%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638217206935590829%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=hPPw%2FTXp%2BTF726NjsAQ3%2B4DxWdeLIih9%2Bc3J%2FwDxcSQ%3D&reserved=0)

AB#24216797

**Comment by rllyy97** [comment]
Are you seeing this in both VSCode and Portal?  Your screenshot appears to be from the VSCode extension but you have the issue labeled as `Standard (Portal)`

**Comment by preetriti1** [comment]
If the connectionid is valid this should not be happening with latest designer which is going in release in portal now

**Comment by ccastrotrejo** [comment]
This is VSCode

**Comment by v-prakumar** [comment]
@ccastrotrejo issue is repro in 2.24.6

**Comment by ccastrotrejo** [comment]
This should be fixed with #2783

**Comment by v-prakumar** [comment]
@ccastrotrejo issue is repro in vsix 2.28.3
![image](https://github.com/Azure/LogicAppsUX/assets/92098490/13c6ce88-566b-42cc-ab1f-5934238dece3)
